rpc/errors.go

```diff
@@ -18,6 +18,11 @@ package rpc
 
 import "fmt"
 
+var (
+	// ErrConnectionClosed is thrown when the connection is closed
+	ErrConnectionClosed = new(connectionClosedError)
```

Comment:
The same error can be thrown in multiple places. For example parsing arguments can throw the same `invalidParamsError` errors when:

1. at least one of the arguments has an invalid type
2. not enough arguments
3. too many arguments

If we introduce a special helper function we must be aware that in all cases the same error code is returned. This is the reason for the special types, to get consistency.